src/Tests/Microsoft.NET.Build.Tests/ReferenceExeTests.cs

```diff
@@ -240,8 +265,12 @@ public void ReferencedExeCanRunWhenReferencesExeWithSelfContainedMismatchForDiff
 
                 project.Root.Element(ns + "PropertyGroup")
                     .Add(XElement.Parse($@"<RuntimeIdentifier Condition=""'$(TargetFramework)' == '{ToolsetInfo.CurrentTargetFramework}'"">" + EnvironmentInfo.GetCompatibleRid() + "</RuntimeIdentifier>"));
+
+                project.Root.Element(ns + "PropertyGroup")
+                    .Add(new XElement(ns + "SelfContained", "true"));
```

Comment:
To test the intended scenario, `SelfContained` should only be set to true for the non `windows` TargetFramework.  Here I've moved both properties into a new PropertyGroup with the appropriate condition.

```suggestion
                var propertyGroup = new XElement(ns + "PropertyGroup",
                    new XAttribute("Condition", $"'$(TargetFramework)' == '{ToolsetInfo.CurrentTargetFramework}'"));

                propertyGroup.Add(new XElement(ns + "RuntimeIdentifier", EnvironmentInfo.GetCompatibleRid()));
                propertyGroup.Add(new XElement(ns + "SelfContained", "true"));

                project.Root.Add(propertyGroup);
```